app/javascript/components/bookmarks/BookmarkManager.jsx

```diff
@@ -0,0 +1,286 @@
+import { Popover, OverlayTrigger } from 'react-bootstrap'
+import BookmarksList from '~/components/bookmarks/BookmarksList'
+import { fetchBookmarks, createBookmark, updateBookmark, deleteBookmark } from '~/lib/scp-api'
+import React, { useEffect, useState } from 'react'
+import _cloneDeep from 'lodash/clone'
+import { isUserLoggedIn } from '~/providers/UserProvider'
+import { useLocation } from '@reach/router'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faLink, faUndo } from '@fortawesome/free-solid-svg-icons'
+import useErrorMessage from '~/lib/error-message'
+
+/**
+ * Component to bookmark views in Explore tab, as well as existing link/reset buttons
+ *
+ * @param bookmarks {Array} existing user bookmark objects
+ * @param clearExploreParams
+ */
+export default function BookmarkManager({bookmarks, clearExploreParams}) {
+  const location = useLocation()
+  const [allBookmarks, setAllBookmarks] = useState(bookmarks)
+  const [saveText, setSaveText] = useState('Save')
+  const [saveDisabled, setSaveDisabled] = useState(false)
+  const [bookmarkSaved, setBookmarkSaved] = useState(null)
+  const [deleteDisabled, setDeleteDisabled] = useState(false)
+  const { ErrorComponent, setShowError, setError } = useErrorMessage()
+  const DEFAULT_BOOKMARK = {
+    name: '',
+    description: '',
+    path: getBookmarkPath()
+  }
+
+  /** copies the url to the clipboard */
+  function copyLink() {
+    navigator.clipboard.writeText(location.href)
+  }
+
+  /** Find a matching bookmark using the current URL params, or return default empty bookmark  */
+  function findExistingBookmark() {
+    const bookmark = allBookmarks.find(bookmark => { return bookmark.path === getBookmarkPath() })
+    if (bookmark) {
+      setBookmarkSaved(true)
+      return bookmark
+    } else {
+      setBookmarkSaved(false)
+      return DEFAULT_BOOKMARK
+    }
+  }
+
+  const [currentBookmark, setCurrentBookmark] = useState(findExistingBookmark)
+
+  /** get bookmark ID from BSON object */
+  function getBookmarkId(bookmark) {
+    return bookmark?._id || bookmark?.id
```

Comment:
Could you remind me what the difference is between `_id` and `id`?

Reply:
Technically, MongoDB IDs are referenced as `_id` in the documents.  However, the Rails convention is to simply call it `id`, though both will return the same thing.  This is fine in the Ruby codebase, but in React we're just dealing with the JSON returned.  While working on this PR I noticed inconsistency in API responses, and I wasn't sure why, so I put that in as a helper.  This was many commits ago, so it may not still be happening.  I'll check it out and if possible refine this code.

Reply:
Some testing has confirmed that API responses will give back `_id`, but our React code is camel-casing this into `id` (not 100% sure where but I think in `scp-api.jsx`) so there's no guarantee of having one over the other at any given time.